src/gradio_demo.py

```diff
@@ -0,0 +1,46 @@
+from domain.ai_model import AiModel
+import gradio as gr
+
+ai_model = AiModel()
+
+
+def detection_from_url(url):
+    det_data = ai_model.process_image_from_url(url, "test_image", True)
+    img_path, det_results, detected_img_path = det_data
+    return img_path, det_results
+
+def classification_from_path(img_path, det_results):
+    if len(det_results["detections"].xyxy) > 0:
+        print("Running classification on detection result(s)...")
+        img_path, classified_animals = ai_model.classify(img_path, det_results)
+        str = f"Found {len(classified_animals)} animal{'s' if len(classified_animals) > 1 else ''}:\n"
+        for animals in classified_animals:
+            animal, confidence = animals['classification']
+            str += f"\t - Class: {animal}, confidence: {confidence*100:.2f}%\n"
```

Comment:
str += is not recommended if the list could be huge. In those cases it's better to collect all strings into a list, and then use ''.join() to build the final string.